server/pkg/asset/status_test.go

```diff
@@ -0,0 +1,177 @@
+package asset
+
+import (
+	"testing"
+
+	"github.com/samber/lo"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestStatus_StatusFrom(t *testing.T) {
+	tests := []struct {
+		Name     string
+		Expected struct {
+			TA   Status
+			Bool bool
+		}
+	}{
+		{
+			Name: "pending",
+			Expected: struct {
+				TA   Status
+				Bool bool
+			}{
+				TA:   StatusPending,
+				Bool: true,
+			},
+		},
+		{
+			Name: "PENDING",
+			Expected: struct {
+				TA   Status
+				Bool bool
+			}{
+				TA:   StatusPending,
+				Bool: true,
+			},
+		},
+		{
+			Name: "in_progress",
+			Expected: struct {
+				TA   Status
+				Bool bool
+			}{
+				TA:   StatusInProgress,
+				Bool: true,
+			},
+		},
+		{
+			Name: "done",
+			Expected: struct {
+				TA   Status
+				Bool bool
+			}{
+				TA:   StatusDone,
+				Bool: true,
+			},
+		},
+		{
+			Name: "failed",
+			Expected: struct {
+				TA   Status
+				Bool bool
+			}{
+				TA:   StatusFailed,
+				Bool: true,
+			},
+		},
+		{
+			Name: "undefined",
+			Expected: struct {
+				TA   Status
+				Bool bool
+			}{
+				TA:   Status(""),
+				Bool: false,
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.Name, func(t *testing.T) {
+			t.Parallel()
+			res, ok := StatusFrom(tc.Name)
+			assert.Equal(t, tc.Expected.TA, res)
+			assert.Equal(t, tc.Expected.Bool, ok)
```

Comment:
DO NOT USE TABLE DRIVEN TESTS when there are few test cases (at least less than 10 test cases).

Maintaining table-driven tests are more painful than you can imagine. Use table-driven tests only if the number of test cases is greater than the pain.

Reply:
Done!